internal/database/database.go

```diff
@@ -515,15 +515,19 @@ func (d *CosmosDBClient) UpdateSubscriptionDoc(ctx context.Context, subscription
 }
 
 // NewCosmosDatabaseClient instantiates a generic Cosmos database client.
-func NewCosmosDatabaseClient(url string, dbName string) (*azcosmos.DatabaseClient, error) {
-	azcoreClientOptions := azcore.ClientOptions{
+func NewCosmosDatabaseClient(url string, dbName string, opts ...func(*azcore.ClientOptions)) (*azcosmos.DatabaseClient, error) {
+	clientOptions := azcore.ClientOptions{
 		// FIXME Cloud should be determined by other means.
 		Cloud: cloud.AzurePublic,
 	}
 
+	for _, o := range opts {
+		o(&clientOptions)
+	}
```

Comment:
I think I'd prefer if `*azcore.ClientOptions` were just passed to the function directly, unless I'm not seeing the advantage to this callback pattern.  Like the "FIXME" comment just above says, at some point we're going to have to stop hard-coding `cloud.AzurePublic` and determine that by other means (e.g. environment variable or IMDS).  I expect to eventually have a centralized function that returns a configured `azcore.ClientOptions`.